### PR TITLE
fix(settings): Fixes Change Pin Setting

### LINKED ISSUE #1630: [Bug]: Pin countdown bypassed when attempting to Change PIN Code
### Describe the bug

For most actions that require entering your PIN, an incorrect entry contributes to the wallet reset countdown. The one option that does not invoke this countdown is "Change PIN Code". This means, if a malicious actor has access to the app, which is quite possible with "Require PIN on launch" disabled, and wants to make a payment but "Require PIN for payments" is enabled, then they have an option of brute-forcing the PIN with infinite attempts and not risk wiping the wallet.

### Reproduce

1. Go to 'Security and Privacy' settings
2. Ensure a PIN code is enabled
3. Attempt to disable the PIN using an incorrect PIN: "x attempts remaining" message is shown and decrements for each wrong attmpt.
4. Attempt to Change PIN Code: Just the "Forgot your PIN?" message is displayed and infinite attempts can be made to guess the PIN. 

### Screenshots / Recording

https://github.com/synonymdev/bitkit/assets/74595920/f8fe9671-dfe6-4271-adbe-35f54e1e177b

### Operating system

Android 13 TKQ1.220829.002

### Bitkit version

v1.0.0-beta.113

### Log output

_No response_

### PR DESCRIPTION
**Pull request by coreyphillips**
### Description
- Enforces pin attempt counter for the "Change Pin" setting.
- Closes #1630

### Linked Issues/Tasks
- #1630

### Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

### Tests
- [x] No test

### QA Notes
Forwards user to Auth screen when attempting to Change Pin in order to enforce the pin counter.

### CODE REVIEW

**Comment by pwltr** [comment]
You have a failing e2e test @coreyphillips

File: src/screens/Settings/Security/index.tsx [hunk 1]
```diff
@@ -128,7 +128,12 @@ const SecuritySettings = ({
 						hide: !pin,
 						testID: 'ChangePIN',
 						onPress: (): void => {
-							navigation.navigate('ChangePin');
+							navigation.navigate('AuthCheck', {
+								onSuccess: () => {
+									navigation.pop();
+									navigation.navigate('ChangePin2');
```
Comment (limpbrains) [inline src/screens/Settings/Security/index.tsx]: Should be `navigation.navigate('ChangePin');`
`ChangePin2` is a second screen where you confirm that your PIN is correct

**Review by limpbrains — CHANGES_REQUESTED** [review]

File: src/screens/Settings/Security/index.tsx (hunk 1 above)
Comment (coreyphillips) [reply src/screens/Settings/Security/index.tsx]: Thanks! Fixed [here](https://github.com/synonymdev/bitkit/pull/1631/commits/2105f508facb42e4ead5dcf19a154d990ef5bac8).

**Comment by coreyphillips** [comment]
> You have a failing e2e test @coreyphillips

Resolved as of [this](https://github.com/synonymdev/bitkit/pull/1631/commits/2105f508facb42e4ead5dcf19a154d990ef5bac8) commit.

**Review by limpbrains — APPROVED** [review]

**Review by pwltr — APPROVED** [review]

**Comment by catch-21** [comment]
@coreyphillips This fixes the problem. Although, the user must now enter their PIN to unlock the option to enter their existing PIN to change their PIN. That's 4 PIN entries now, it was 3. Would it be possible to add the countdown to the existing PIN entry to avoid the extra step?

**Pull request merged**